src/sql/src/catalog.rs

```diff
@@ -262,20 +263,16 @@ impl fmt::Display for CatalogError {
             Self::UnknownDatabase(name) => write!(f, "unknown database '{}'", name),
             Self::UnknownSchema(name) => write!(f, "unknown schema '{}'", name),
             Self::UnknownItem(name) => write!(f, "unknown catalog item '{}'", name),
-            Self::InvalidSinkDependency(name) => write!(
+            Self::InvalidDependency { name, typ } => write!(
                 f,
-                "catalog item '{}' is a sink and so cannot be depended upon",
-                name
-            ),
-            Self::InvalidIndexDependency(name) => write!(
-                f,
-                "catalog item '{}' is an index and so cannot be depended upon",
-                name
-            ),
-            Self::InvalidTypeDependency(name) => write!(
-                f,
-                "catalog item '{}' is a type and so cannot be depended upon",
-                name
+                "catalog item '{}' is {} {} and so cannot be depended upon",
+                name,
+                if matches!(typ, CatalogItemType::Index) {
+                    "an"
+                } else {
+                    "a"
+                },
```

Comment:
Grammar is the worst!

Reply:
Or is the problem that computers are too simple to keep up? 🤔